recipes/catalyst/all/conanfile.py

```diff
@@ -0,0 +1,86 @@
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps, cmake_layout
+from conan.tools.files import get, copy, rmdir
+import os
+
+required_conan_version = ">=1.53.0"
+
+
+class CatalystConan(ConanFile):
+    name = "catalyst"
+    package_type = "library"
+    description = "Catalyst is an API specification developed for simulations (and other scientific data producers) to analyze and visualize data in situ."
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://gitlab.kitware.com/paraview/catalyst"
+    topics = ("simulation", "visualization", "paraview", "in-situ", "in-transit")
+    license = "BSD-3"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "use_mpi": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "use_mpi": False,
+    }
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>3.26]")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def configure(self):
+        if self.options.get_safe("shared"):
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CATALYST_BUILD_REPLAY"] = False
+        tc.variables["CATALYST_BUILD_TESTING"] = False
+        tc.variables["CATALYST_BUILD_SHARED_LIBS"] = self.options.shared
+        tc.variables["CATALYST_USE_MPI"] = self.options.use_mpi
```

Comment:
You should also probably add a conditional `self.requires("openmpi/x.y.z")`.
The OpenMPI recipe is currently not yet migrated to Conan 2.0, but you can give this version a try: #18980.

Reply:
If its okay, I will remove the option to use MPI and wait until the openmpi recipe is ported.

Reply:
I would leave the option and add a `if self.options.use_mpi: raise ConanException("MPI is not yet supported by the recipe")` to `def validate(self)`. Plus a TODO comment mentioning that OpenMPI is not yet Conan 2.0-compatible.